src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -200,25 +249,53 @@ public boolean removePerson(ReadOnlyPerson key) throws PersonNotFoundException {
 
     /**
      * Updates {@code key} to exclude {@code key} from the blacklist in this {@code AddressBook}.
+     * @return ReadOnly {@code newUnBlacklistedPerson}.
      * @throws PersonNotFoundException if the {@code key} is not in this {@code AddressBook}.
      */
-    public void removeBlacklistedPerson(ReadOnlyPerson key) throws PersonNotFoundException {
+    public ReadOnlyPerson removeBlacklistedPerson(ReadOnlyPerson key) throws PersonNotFoundException {
         int index;
         index = persons.getIndexOf(key);
 
-        Person newBlacklistedPerson = new Person(key);
-        newBlacklistedPerson.setIsBlacklisted(false);
+        Person newUnBlacklistedPerson = new Person(key);
+        newUnBlacklistedPerson.setIsBlacklisted(false);
+
+        if (newUnBlacklistedPerson.getDebt().toNumber() == 0) {
+            newUnBlacklistedPerson.setIsWhitelisted(true);
+        }
+
         persons.remove(key);
-        syncMasterTagListWith(newBlacklistedPerson);
-        // TODO: the tags master list will be updated even though the below line fails.
-        // This can cause the tags master list to have additional tags that are not tagged to any person
-        // in the person list.
+
         try {
-            persons.add(index, newBlacklistedPerson);
+            persons.add(index, newUnBlacklistedPerson);
         } catch (DuplicatePersonException e) {
-            assert false : "This is not possible as prior checks have been done";
+            assert false : "This is not possible as prior checks have"
+                    + " been done to ensure AddressBook does not have duplicate persons";
         }
+
+        return persons.getReadOnlyPerson(index);
     }
+
+    /**
+     * Updates {@code key} to exclude {@code key} from the whitelist in this {@code AddressBook}.
+     * @return ReadOnly {@code newWhitelistedPerson}.
```

Comment:
same as abv. One way you can check whether you've done it correctly is to place ur cursor at the method name then `ctrl-Q`. It wil open up the documentation for that method and you can check the formatting.

Reply:
Woah that's so coool! Thanks for the tip Jel!